src/server/ServerHandler.java

```diff
@@ -304,6 +306,62 @@ private Response handleLogin(User user) {
         }
     }
 
+    /**
+     * Handles acknowledge response request after an invitation response
+     *
+     * @param eventId The eventId of the initial invitation
+     * @return Response indicating success or failure of acknowledgment
+     */
+    private Response handleAcknowledgeResponse(int eventId) {
+        try {
+            // Use the database helper function getEvent() to retrieve the Event object
+            Event event = DatabaseHelper.getInstance().getEvent(eventId);
+
+            // Case 1: Check if the event exists, and if the sender of the event is the current username
+            if (event == null) {
+                return new Response(ResponseStatus.FAILURE, "Event with ID " + eventId + " not found.");
+            }
+
+            if (!event.getSender().equals(this.currentUsername)) {
+                return new Response(ResponseStatus.FAILURE, "You are not the sender of this invitation.");
+            }
+
+            // Get the current status of the event
+            Event.EventStatus currentStatus = event.getStatus();
+
+            // Case 2: If the response was DECLINED, set the status to ABORTED
+            if (currentStatus == Event.EventStatus.DECLINED) {
+                event.setStatus(Event.EventStatus.ABORTED);
+                DatabaseHelper.getInstance().updateEvent(event);
+                return new Response(ResponseStatus.SUCCESS, "Game invitation declined and aborted successfully.");
+            }
+            // Case 3: If the response was ACCEPTED
+            else if (currentStatus == Event.EventStatus.ACCEPTED) {
+                // Set currentEventId to eventId
+                this.currentEventId = eventId;
+
+                // Abort any other pending invitation the user might have from other players
+                DatabaseHelper.getInstance().abortAllUserEvents(this.currentUsername);
+
+                // Update the event status to PLAYING
+                event.setStatus(Event.EventStatus.PLAYING);
+                DatabaseHelper.getInstance().updateEvent(event);
+
+                return new Response(ResponseStatus.SUCCESS, "Game invitation accepted! Game is now starting.");
+            }
+            else {
```

Comment:
else should be on the same line as the closing brace
```suggestion
            } else {
```